subprojects/core/src/main/java/org/gradle/execution/plan/ConsumerState.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.execution.plan;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class ConsumerState {
```

Comment:
💅 New top-level class needs javadoc, maybe `final`?

Reply:
Made final. 

Unfortunately I'm not sure I know all that much about this type and its usage. I just moved the code somewhere else